geopandas/tests/test_crs.py

```diff
@@ -336,6 +336,44 @@ def test_dataframe_setitem(self):
         assert df["geometry"].crs == self.wgs
         assert df["other_geom"].crs == self.osgb
 
+    def test_dataframe_multiindex_cols_2level(self):
+        # GH1763 https://github.com/geopandas/geopandas/issues/1763
```

Comment:
Added a second test for n>2 because behaviour is actually slightly different. Prior to the changes above, with 2 levels in the multiindex, `gdf.geometry` is assigned to None, but with 3 levels, it is set correctly (but the geometry column is still messed up. Despite that the tests look basically the same, so one could be dropped (or at the very least parametrised to avoid duplication, but I'll wait to see if the n=3 case is needed)

Reply:
I have hit this one in the wild with n=3 and not with n=2 so I'd keep the two as they are. I don't think it needs to be parametrised, given we have only 2 options... Unless you want to parametrically test for N(2...X) cases.